internal/report/manager.go

```diff
@@ -0,0 +1,162 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package report
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Manager represents a structure that manages all of the different reporters
+type Manager struct {
+	client    Client
+	reporters map[string]Reporter
```

Comment:
I'm not sure I understand this part where we have many reporters. It seems like we only care about the `snapshot` one. We never call Report on any other kind, and all the processors only have access to the `snapshot` Reporter. Is there a design reason for this? Seems overly abstract for what we end up actually doing.

Reply:
I originally designed this thinking we would also report throughput via a reporter. We haven't fully implemented that feature set yet so it's not off the table. I also think this leaves a fairly easy door open to adding other types of reporters. I think as BPOP evolves we are going to extract more data out of the collector and the reporter structure allows that. 

I am in favor revisiting this in a quarter or two and if it hasn't been expanded upon then we should condense it to the one use case to reduce complexity.